sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -586,26 +638,81 @@ public HttpPipeline getHttpPipeline() {
     public Mono<Response<SendMessageResult>> sendMessageWithResponse(String messageText, Duration visibilityTimeout,
                                                                    Duration timeToLive) {
         try {
-            return withContext(context -> sendMessageWithResponse(messageText, visibilityTimeout, timeToLive, context));
+            BinaryData message = messageText == null ? null : BinaryData.fromString(messageText);
```

Comment:
When `messageText` is empty do we want null `BinaryData` or an empty `BinaryData`? `BinaryData.fromString` is null safe and will return an empty `BinaryData`

Reply:
This should be fine. In tests I discovered that previously passing "null" resulted in "XML format exception or so" so producing empty binarydata should be fine.

Reply:
Other thing is that when message ends up being enqueued on the service side then worst case is service returning empty string at peek/receive.